src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -266,14 +267,12 @@ public void getDocIds(DocIdPusher pusher) throws IOException,
   void addMetadataToRecordBuilder(DocIdPusher.Record.Builder builder,
       ResultSet rs) throws SQLException {
     ResultSetMetaData rsMetaData = rs.getMetaData();
-    int numberOfColumns = rsMetaData.getColumnCount();
-    for (int i = 1; i < (numberOfColumns + 1); i++) {
-      String columnName = rsMetaData.getColumnLabel(i);
-      Object value = rs.getObject(i);
-      String key = metadataColumns.getMetadataName(columnName);
-      if (key != null) {
-        builder.addMetadata(key, "" + value);
-      }
+    MetadataColumns metadataCols = (metadataColumns == null)
+        ? new MetadataColumns(rsMetaData) : metadataColumns;
+    for (String columnName : metadataCols.keySet()) {
```

Comment:
That would skip any *magic* that happens in getMetadataName(), as was the case in AllColumns.  In reality, MetadataColumns can be replaced with a Map and a couple of factory methods in a separate refactoring CL.